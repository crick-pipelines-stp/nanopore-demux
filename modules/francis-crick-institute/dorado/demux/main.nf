process DORADO_DEMUX {
    tag "$meta.id"
    label 'process_high'

    container 'docker.io/thecrick/pipetech_dorado:0.7.3-linux-x64'

    input:
    tuple val(meta), path(bam)
    val emit_bam

    output:
    tuple val(meta), path("*.bam")     , emit: bam , optional: true
    tuple val(meta), path("*.fastq.gz"), emit: fastq , optional: true
    path  "versions.yml"               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def emit = emit_bam ? '' : '--emit-fastq'

    """
    dorado demux \\
        --output-dir ./ \\
        --no-classify \\
        $emit \\
        $args \\
        $bam

    if ls ./*.fastq 1> /dev/null 2>&1; then
        for file in ./*.fastq; do
            gzip \"\$file\"
        done
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dorado: \$(dorado --version 2>&1)
    END_VERSIONS
    """
}
