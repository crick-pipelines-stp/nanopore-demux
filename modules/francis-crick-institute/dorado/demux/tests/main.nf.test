nextflow_process {

    name "Test Module DORADO_DEMUX"
    script "../main.nf"
    process "DORADO_DEMUX"

    tag "modules"
    tag "ont"
    tag "ont_dorado"
    tag "ont_dorado/demux"

    test("fastq") {
        tag "ont_dorado/demux/fastq"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/1000_reads_6_samples.bam', checkIfExists: true)
                ])
                input[1] = false
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match()}
            )
        }
    }

    test("bam") {
        tag "ont_dorado/demux/bam"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/multiplexed/1000_reads_6_samples.bam', checkIfExists: true)
                ])
                input[1] = true
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match()}
            )
        }
    }

}
