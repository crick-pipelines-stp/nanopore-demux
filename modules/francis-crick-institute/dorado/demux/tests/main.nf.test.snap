{
    "fastq": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "barcode01.fastq.gz:md5,a6f60d91a0e6cdac1f3bcd79415a2d6d",
                            "barcode02.fastq.gz:md5,db3213cc6685a4379f00aad0718fc3e3",
                            "barcode03.fastq.gz:md5,8d6e8229a3bcbb1f9aa3bcf9821e653d",
                            "barcode04.fastq.gz:md5,c2fa3323177e508c7a0c37c7c3bcdae4",
                            "barcode05.fastq.gz:md5,b9ab5f9c808b970664e95833c12d2ffe",
                            "barcode06.fastq.gz:md5,7a0cad115c6acac739cb7d884e442824"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,8376afad0a2e464d10badd604ba1315a"
                ],
                "bam": [
                    
                ],
                "fastq": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "barcode01.fastq.gz:md5,a6f60d91a0e6cdac1f3bcd79415a2d6d",
                            "barcode02.fastq.gz:md5,db3213cc6685a4379f00aad0718fc3e3",
                            "barcode03.fastq.gz:md5,8d6e8229a3bcbb1f9aa3bcf9821e653d",
                            "barcode04.fastq.gz:md5,c2fa3323177e508c7a0c37c7c3bcdae4",
                            "barcode05.fastq.gz:md5,b9ab5f9c808b970664e95833c12d2ffe",
                            "barcode06.fastq.gz:md5,7a0cad115c6acac739cb7d884e442824"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8376afad0a2e464d10badd604ba1315a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:31:10.999587"
    },
    "bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "barcode01.bam:md5,d0b28dace9bdc1ae921b0e0d092239d1",
                            "barcode02.bam:md5,8a9bd32509eab5c3989624d2109feae5",
                            "barcode03.bam:md5,5289f36ec2fa8bda382b47031ee77db8",
                            "barcode04.bam:md5,eb9a89c095a9cae2808c1145337e88aa",
                            "barcode05.bam:md5,7dfa0c31f8a7c648fbcb371c5afa9541",
                            "barcode06.bam:md5,03b2229ab201fa51f3ee8f3c6990e9e8"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,8376afad0a2e464d10badd604ba1315a"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "barcode01.bam:md5,d0b28dace9bdc1ae921b0e0d092239d1",
                            "barcode02.bam:md5,8a9bd32509eab5c3989624d2109feae5",
                            "barcode03.bam:md5,5289f36ec2fa8bda382b47031ee77db8",
                            "barcode04.bam:md5,eb9a89c095a9cae2808c1145337e88aa",
                            "barcode05.bam:md5,7dfa0c31f8a7c648fbcb371c5afa9541",
                            "barcode06.bam:md5,03b2229ab201fa51f3ee8f3c6990e9e8"
                        ]
                    ]
                ],
                "fastq": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8376afad0a2e464d10badd604ba1315a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:31:13.587299"
    }
}